src/core/ext/filters/http/message_compress/message_compress_filter.c

```diff
@@ -89,7 +90,8 @@ static bool skip_compression(grpc_call_element *elem, uint32_t flags,
     return 1;
   }
   if (has_compression_algorithm) {
-    if (calld->compression_algorithm == GRPC_COMPRESS_NONE) {
+    if (calld->compression_algorithm == GRPC_COMPRESS_NONE ||
+        GRPC_IS_STREAM_COMPRESSION_ALGORITHM(calld->compression_algorithm)) {
```

Comment:
They are. The original idea of combining them in one enum is to disallowing both of them appearing together, but you are right that separating them would be much better in term of code. I will come up with another way of this mutual exclusion.